checks/frozen_deps.go

```diff
@@ -35,19 +42,113 @@ func FrozenDeps(c *checker.CheckRequest) checker.CheckResult {
 	return checker.MultiCheckAnd(
 		isPackageManagerLockFilePresent,
 		isGitHubActionsWorkflowPinned,
+		isDockerfilePinned,
 	)(c)
 }
 
-// TODO(laurent): need to support Docker https://github.com/ossf/scorecard/issues/403
 // TODO(laurent): need to support GCB
 
+// ============================================================
+// ======================== Dockerfiles =======================
+// ============================================================
+func isDockerfilePinned(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(frozenDepsStr, "*Dockerfile*", false, c, validateDockerfile)
+}
+
+func isPresent(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func validateDockerfile(path string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	// Users may use various names, e.g.,
+	// Dockerfile.aarch64, Dockerfile.template, Dockerfile_template, dockerfile, Dockerfile-name.template
+	// Templates may trigger false positives, e.g. FROM { NAME }.
+
+	// We have what looks like a docker file.
+	// Let's interpret the content as utf8-encoded strings.
+	contentReader := strings.NewReader(string(content))
+	regex := regexp.MustCompile(`.*@sha256:[a-f\d]{64}`)
+
+	r := true
```

Comment:
nit: `r` is used quite far away from where it is defined, so it's more helpful to rename this something more meaningful, like `pinned`.